Detectors/MUON/MID/Raw/src/ELinkDataShaper.cxx

```diff
@@ -43,27 +43,33 @@ ELinkDataShaper::ELinkDataShaper(bool isDebugMode, bool isLoc, uint8_t uniqueId,
   mLocalToBCSelfTrig = mElectronicsDelay.localToBC;
   if (!isLoc) {
     mLocalToBCSelfTrig -= electronicsDelay.localToReg;
-    mElectronicsDelay.calibToFET += electronicsDelay.localToReg;
   }
+  mElectronicsDelay.calibToFET += mLocalToBCSelfTrig;
 }
 
-void ELinkDataShaper::set(uint32_t orbit)
+void ELinkDataShaper::set(uint32_t orbit, uint32_t trigger)
 {
   /// Sets the orbit and the output data vectors
-  mRDHOrbit = orbit;
 
   if (mIR.isDummy()) {
+    // First initialization
     mIR.bc = 0;
-// The reset changes depending on the way we synch with the orbit
-// (see processOrbitTrigger for details)
-// FIXME: pick one of the two
-#ifdef ORBITFROMMIDRO
-    mIR.orbit = orbit - 1; // with orbit increase
-#else
-    mIR.orbit = orbit;   // with reset to RDH
-#endif
+    mIR.orbit = orbit - 1;
```

Comment:
what happens in the boundary case where orbit=0 ? Is that guaranteed to never happen in this part of the code ?